### PR TITLE
Feat: Cloud Free Deprecation test case

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
- Epic : https://mattermost.atlassian.net/browse/MM-51411

### CODE REVIEW

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+8. Click on `Upgrade` button on `Professional` card.
+9. Verify the `Purchase` modal is opened
+10. Enter `Card` details, `Billing Address` and `Seats`
+11. Click on `Upgrade` button on the RHS
+12. Verify the Upgrade is successful and license is applied. (Navigate to channels and verify `view plan` is not be shown on the top right of banner)
+13. Verify the Billing History page to confirm purchase details
+
+Scenario 2: When instance is 10 days
+
+1. Repeat above steps but verify that Global banner and CTA now has red background color.
```
Comment (AGMETEOR) [inline data/test-cases/cloud/cloud-free-deprecation/test-1.md:57]: Do we also want to test scenario of the dismissable banner reappearing, (after dismissal) at the 60, 30, 10 days to deprecation interval

Comment (yasserfaraazkhan) [reply data/test-cases/cloud/cloud-free-deprecation/test-1.md:57]: sure Allan, I'll add a step for it.

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+Ref Epic: <https://mattermost.atlassian.net/browse/MM-51411>
+
+Scenario 1: When instance is 90 days (or 60 days or 30 days), successfully upgrade to a paid plan
+
+1. Login with Admin account
+2. Verify Global Banner with a message 'Cloud Free will be depreicated on MM DD YYY. To keep your workspace, upgrade to a paid plan.'
+3. Navigate to System console > Subscriptions
+4. Verify a CTA is displayed with a message 'Upgrade to a paid plan to keep your workspace'
+5. Verify `Upgrade` and `Contact Sales` button is clickable
+6. When User clicks on `Contact Sales` button, verify Zendesk form is opened in a new Tab
```
Comment (johndavidlugtu) [inline data/test-cases/cloud/cloud-free-deprecation/test-1.md:46]: Actually, any "Contact Sales" button should redirect to mattermost.com/contact-sales

"Contact support" will be the one that redirects to Zendesk

https://mattermost.atlassian.net/browse/MM-51682

**Review by johndavidlugtu — CHANGES_REQUESTED** [review]
@yasserfaraazkhan good stuff so far!

We'll also want to verify banners inside the system console are also visible.

Also, can we also make add to verify everything mentioning "Cloud Free" and not "Cloud Starter"? Thanks

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md (hunk 1 above)
Comment (AGMETEOR) [reply data/test-cases/cloud/cloud-free-deprecation/test-1.md:57]: There's another test case for the changes in the pricing modal here https://github.com/mattermost/mattermost-server/pull/22639

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md (hunk 2 above)
Comment (yasserfaraazkhan) [reply data/test-cases/cloud/cloud-free-deprecation/test-1.md:46]: Sure John, I've updated the changes

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md (hunk 1 above)
Comment (yasserfaraazkhan) [reply data/test-cases/cloud/cloud-free-deprecation/test-1.md:57]: Thank you @AGMETEOR , I've added a test for this

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+---
+
+**Step 1**
+
+Ref Epic: <https://mattermost.atlassian.net/browse/MM-51411>
+
+Scenario 1: When instance is 90 days (or 60 days or 30 days), successfully upgrade to a paid plan
+
+1. Login with Admin account
+2. Verify Global Banner with a message 'Cloud Free will be depreicated on MM DD YYY. To keep your workspace, upgrade to a paid plan.'
```
Comment (lindalumitchell) [inline data/test-cases/cloud/cloud-free-deprecation/test-1.md:42]:
```suggestion
2. Verify Global Banner with a message 'Cloud Free will be deprecated on MM DD YYY. To keep your workspace, upgrade to a paid plan.'
```

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+3. Navigate to System console > Subscriptions
```
Comment (lindalumitchell) [inline data/test-cases/cloud/cloud-free-deprecation/test-1.md:43]:
```suggestion
3. Navigate to System Console > Subscriptions
```

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+4. Verify a CTA is displayed with a message 'Upgrade to a paid plan to keep your workspace'
+5. Verify `Upgrade` and `Contact Sales` button is clickable
+6. When User clicks on `Contact Sales` button, verify a new tab opens with `https://mattermost.com/contact-sales/`
+7. Clicks on the `Upgrade` button, verify Plans modal is opened
```
Comment (lindalumitchell) [inline data/test-cases/cloud/cloud-free-deprecation/test-1.md:47]:
```suggestion
7. Click on the `Upgrade` button, verify Plans modal is opened
```

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+8. Click on `Upgrade` button on `Professional` card.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/cloud-free-deprecation/test-1.md:48]:
```suggestion
8. Click on `Upgrade` button on `Professional` card
```

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+9. Verify the `Purchase` modal is opened
+10. Enter `Card` details, `Billing Address` and `Seats`
+11. Click on `Upgrade` button on the RHS
+12. Verify the Upgrade is successful and license is applied. (Navigate to channels and verify `view plan` is not be shown on the top right of banner)
```
Comment (lindalumitchell) [inline data/test-cases/cloud/cloud-free-deprecation/test-1.md:52]:
```suggestion
12. Verify the Upgrade is successful and license is applied. (Navigate to channels and verify `view plan` is not shown on the top right of banner)
```

File: data/test-cases/cloud/cloud-free-deprecation/test-1.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+13. Verify the Billing History page to confirm purchase details
+
+Scenario 2: When instance is 10 days
+
+1. Repeat above steps but verify that Global banner and CTA now has red background color.
+
+Scenario 3: Dismissed Global Banners should reappear after Admin
```
Comment (lindalumitchell) [inline data/test-cases/cloud/cloud-free-deprecation/test-1.md:59]:
```suggestion
Scenario 3: Dismissed Global Banners should reappear after Admin dismisses and logs out and back in
```

**Review by lindalumitchell — APPROVED** [review]
Approving, with just a few very minor suggestions. :)

**Pull request merged**